x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -252,6 +253,23 @@ public void deleteResponse(AsyncExecutionId asyncExecutionId,
         return asyncTask;
     }
 
+    /**
+     * Returns the {@link AsyncTask} if the provided <code>asyncTaskId</code>
+     * is registered in the task manager, <code>null</code> otherwise.
+     *
+     */
+    <T extends AsyncTask> T getTaskStatus(TaskManager taskManager, AsyncExecutionId asyncExecutionId, Class<T> tClass) {
```

Comment:
Can you add an option on `getTask` instead to enable/disable the authentication  check ?

Reply:
Thanks for the comment, @jimczi .  I have talked to @albertzaharovits about this sometime ago, and he suggested me from a security point of view to use a separate method instead of adding a parameter to enable/disable authentication. 

I ended up  moving the logic of this method to `TransportGetAsyncStatusAction` in order to avoid parametrized function arguments.